New config parameter: executable

If I build a project with Java 8 I get an exception:
```
--- maven-jpublisher-plugin:1.0.0:generate (default) @ sznyr-leksz ---
Generating JPublisher classes...
Exception in thread "main" java.lang.NoClassDefFoundError: sun/io/CharToByteConverter
	at sqlj.util.io.UCode_CharStream.<init>(UCode_CharStream.java:169)
	at sqlj.util.io.UCode_CharStream.<init>(UCode_CharStream.java:187)
	at oracle.jpub.Doit.mainSql(Doit.java:245)
	at oracle.jpub.Doit.main(Doit.java:126)
	at oracle.jpub.Doit.main(Doit.java:101)
	at oracle.jpub.Doit.main(Doit.java:57)
	at oracle.jpub.publish.Main.main(Main.java:9)
	at oracle.jpub.Main.main(Main.java:6)
Caused by: java.lang.ClassNotFoundException: sun.io.CharToByteConverter
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 8 more
Generation finished
```

May I set javac in config with __executable__ property?
`<executable>${java.path}/javac</executable>`

--- a/pom.xml
+++ b/pom.xml
@@ -10,7 +10,7 @@
 
     <groupId>hu.vanio.maven.plugins</groupId>
     <artifactId>maven-jpublisher-plugin</artifactId>
-    <version>1.0.2-SNAPSHOT</version>
+    <version>1.1.0-SNAPSHOT</version>
     <packaging>maven-plugin</packaging>
 
     <name>Maven JPublisher Plugin</name>

--- a/src/main/java/hu/vanio/maven/plugins/jpublisher/JpublisherGenerateMojo.java
+++ b/src/main/java/hu/vanio/maven/plugins/jpublisher/JpublisherGenerateMojo.java
@@ -73,7 +73,7 @@ public class JpublisherGenerateMojo extends AbstractMojo {
      * Sets the executable of the compiler to use when {@link #fork} is
      * <code>true</code>.
      */
-    @Parameter(property = "maven.compiler.executable")
+    @Parameter
     private String executable;
 
     /**
@@ -147,10 +147,12 @@ public class JpublisherGenerateMojo extends AbstractMojo {
         this.getLog().debug("Classpath: " + classPath);
 
         StringBuilder jpubCommand = new StringBuilder();
-        if (executable != null) {
+        if (executable == null) {
             jpubCommand.append(javaHome).append("/bin/java ");
+            this.getLog().info("Java path: " + jpubCommand);
         } else {
-            jpubCommand.append(executable);
+            jpubCommand.append(executable).append(" ");
+            this.getLog().info("Java path: " + jpubCommand);
         }
 
         jpubCommand.append(String.format("-cp %s ", classPath));
@@ -161,7 +163,7 @@ public class JpublisherGenerateMojo extends AbstractMojo {
         jpubCommand.append(String.format("-props=%s ", propsFile));
         jpubCommand.append(String.format("-input=%s ", typeListFile));
 
-        this.getLog().debug("Starting process: " + jpubCommand);
+        this.getLog().info("Starting process: " + jpubCommand);
 
         Process p = Runtime.getRuntime().exec(jpubCommand.toString());
         readAndPrintStream(p.getInputStream());

--- a/src/main/java/hu/vanio/maven/plugins/jpublisher/JpublisherGenerateMojo.java
+++ b/src/main/java/hu/vanio/maven/plugins/jpublisher/JpublisherGenerateMojo.java
@@ -34,12 +34,13 @@ import org.apache.maven.project.MavenProject;
 import org.codehaus.plexus.util.FileUtils;
 
 /**
- * This goal calls Oracle JPublisher to generate Java classes from Oracle SQL types and procedures
+ * This goal calls Oracle JPublisher to generate Java classes from Oracle SQL
+ * types and procedures
  *
  * @author Gyula Szalai <gyszalai@gmail.com>
  */
-@Mojo(name = "generate", defaultPhase = LifecyclePhase.GENERATE_SOURCES, 
-      threadSafe = true, requiresDependencyResolution = ResolutionScope.COMPILE_PLUS_RUNTIME)
+@Mojo(name = "generate", defaultPhase = LifecyclePhase.GENERATE_SOURCES,
+        threadSafe = true, requiresDependencyResolution = ResolutionScope.COMPILE_PLUS_RUNTIME)
 public class JpublisherGenerateMojo extends AbstractMojo {
 
     /** Classpath separator */
@@ -68,6 +69,12 @@ public class JpublisherGenerateMojo extends AbstractMojo {
     /** JPublisher typelist file (-input command line switch) */
     @Parameter(defaultValue = "${project.basedir}/src/main/jpublisher/jpub-typelist.txt")
     private String typeListFile;
+    /**
+     * Sets the executable of the compiler to use when {@link #fork} is
+     * <code>true</code>.
+     */
+    @Parameter(property = "maven.compiler.executable")
+    private String executable;
 
     /**
      * Main entry point
@@ -79,7 +86,7 @@ public class JpublisherGenerateMojo extends AbstractMojo {
             getLog().info("JPublisher code generation has been skipped per configuration of the skip parameter.");
             return;
         }
-        
+
         String genSourceRoot = project.getBasedir().toString() + "/target/generated-sources/jpublisher";
         FileUtils.mkdir(genSourceRoot);
         project.addCompileSourceRoot(genSourceRoot);
@@ -101,9 +108,11 @@ public class JpublisherGenerateMojo extends AbstractMojo {
     }
 
     /**
-     * Creates a classpath string from runtime dependencies of the active project
+     * Creates a classpath string from runtime dependencies of the active
+     * project
      * @return The generated classpath string
-     * @throws MojoExecutionException If getting runtime dependencies fails for any reason
+     * @throws MojoExecutionException If getting runtime dependencies fails for
+     * any reason
      */
     private String getRuntimeClasspath() throws MojoExecutionException {
         StringBuilder sb = new StringBuilder();
@@ -128,16 +137,22 @@ public class JpublisherGenerateMojo extends AbstractMojo {
 
     /**
      * Starts the JPublisher generator process
-     * @param genSourceRoot The source root where the generated sources will be generated
+     * @param genSourceRoot The source root where the generated sources will be
+     * generated
      */
     private void startJpublisherProcess(String genSourceRoot) throws IOException, InterruptedException, MojoExecutionException, DependencyResolutionRequiredException {
-        
+
         String javaHome = System.getenv("JAVA_HOME");
         String classPath = this.getRuntimeClasspath();
         this.getLog().debug("Classpath: " + classPath);
 
         StringBuilder jpubCommand = new StringBuilder();
-        jpubCommand.append(javaHome).append("/bin/java ");
+        if (executable != null) {
+            jpubCommand.append(javaHome).append("/bin/java ");
+        } else {
+            jpubCommand.append(executable);
+        }
+
         jpubCommand.append(String.format("-cp %s ", classPath));
         jpubCommand.append("oracle.jpub.Main ");
         jpubCommand.append(String.format("-url=%s ", dbUrl));
@@ -145,7 +160,7 @@ public class JpublisherGenerateMojo extends AbstractMojo {
         jpubCommand.append(String.format("-dir=%s ", genSourceRoot));
         jpubCommand.append(String.format("-props=%s ", propsFile));
         jpubCommand.append(String.format("-input=%s ", typeListFile));
-        
+
         this.getLog().debug("Starting process: " + jpubCommand);
 
         Process p = Runtime.getRuntime().exec(jpubCommand.toString());
@@ -155,7 +170,7 @@ public class JpublisherGenerateMojo extends AbstractMojo {
 
         this.getLog().debug("Exit value: " + exitValue);
     }
-    
+
     private void readAndPrintStream(InputStream is) throws IOException {
         BufferedReader reader = new BufferedReader(new InputStreamReader(is));
         try {
@@ -167,5 +182,5 @@ public class JpublisherGenerateMojo extends AbstractMojo {
             reader.close();
         }
     }
-    
+
 }

